google/auth/transport/grpc.py

```diff
@@ -149,3 +154,44 @@ def secure_authorized_channel(
     )
 
     return grpc.secure_channel(target, composite_credentials, **kwargs)
+
+
+class SslCredentials:
+    """Class for application default SSL credentials. For Linux with endpoint
+    verification support, device certificate will be automatically loaded if
+    available and mutual TLS will be established.
+    """
+
+    def __init__(self):
+        self._is_mtls = False
+
+        # Load client SSL credentials.
+        context_aware_metadata = _mtls_helper._read_dca_metadata_file(
+            _mtls_helper.CONTEXT_AWARE_METADATA_PATH
+        )
+        if context_aware_metadata:
+            try:
+                cert, key = _mtls_helper.get_client_ssl_credentials(
+                    context_aware_metadata
+                )
+                self._is_mtls = True
+
+            except (NotImplementedError, OSError, RuntimeError, ValueError) as e:
+                _LOGGER.debug(
```

Comment:
I don't think these errors should be caught and discarded here.

If you want consistency with the Go implementation, the call to get_client_ssl_credentials would be moved to the ssl_credentials method.

Then _is_mtls = True if context_aware_metadata != None

Reply:
done